### PR TITLE
Feature/lottery nft

### PR DESCRIPTION
**Pull request by ChefHutch**
Preview: https://lottery-nft.netlify.app/collectibles

### Prior to launch:

- [ ] Needs mainnet contract deploying and then adding to `contracts` - Doge/Nyan to deploy, Hachi or Cheems to add to PR
- [x] Community testing [_in progress_] - Nina to ✅ 

_note - achievements haven't been tested_

### CODE REVIEW

File: src/views/Collectibles/hooks/useBunnySpecialLottery.tsx [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+  const genericClaimChecks = useCallback(
+    async (variationId: number | string): Promise<boolean> => {
+      let hasClaimed = false
+      if (!userRounds) {
+        return false
+      }
+      try {
+        hasClaimed = await lotteryNftContract.hasClaimed(account, variationId)
+      } catch (error) {
+        console.error(`Failed to check hasClaimed for id ${variationId}`, error)
```
Comment (hachiojidev) [inline src/views/Collectibles/hooks/useBunnySpecialLottery.tsx:111]: What happens if this error is thrown? Maybe return true or false?

Comment (ChefHutch) [reply src/views/Collectibles/hooks/useBunnySpecialLottery.tsx:111]: Nice catch 👍

File: src/views/Collectibles/index.tsx [hunk 2]
```diff
@@ -13,6 +15,12 @@ const StyledHero = styled.div`
 
 const Collectibles = () => {
   const { t } = useTranslation()
+  const dispatch = useAppDispatch()
+
+  useEffect(() => {
+    // get current lottery ID - used for lottery NFT checks
+    dispatch(fetchCurrentLotteryId())
+  }, [dispatch])
```
[the remark this answers is not in the document]
Comment (ChefHutch) [reply src/views/Collectibles/index.tsx:23]: Ah - actually, great flag. I will need to move this to the `useBunnySpecialLottery` hook as the global check could be fired from any page and will need the currentLotteryId.

File: src/views/Collectibles/components/ClaimNftModal.tsx [hunk 3]
```diff
@@ -30,14 +30,16 @@ const ClaimNftModal: React.FC<ClaimNftModalProps> = ({ nft, onSuccess, onClaim,
...
 
   const handleConfirm = async () => {
-    const tx = await onClaim()
     setIsConfirming(true)
-    const receipt = await tx.wait()
-    if (receipt.status) {
-      toastSuccess(t('Successfully claimed!'))
-      onDismiss()
-      onSuccess()
-    } else {
```
Comment (ChefHutch) [inline src/views/Collectibles/components/ClaimNftModal.tsx:40]: In this previous pattern the `else` condition isn't hit when the tx fails, so the fail state functions never trigger.

File: src/views/Collectibles/components/GlobalCheckClaimStatus.tsx [hunk 4]
```diff
@@ -18,35 +19,56 @@ interface GlobalCheckClaimStatusProps {
...
   const { account } = useWeb3React()
   const { pathname } = useLocation()
+  const { canClaimBaller, canClaimLottie, canClaimLucky } = useBunnySpecialLottery()
 
   // Check claim status
   useEffect(() => {
     const fetchClaimStatus = async () => {
-      const canClaim = await easterNftContract.canClaim(account)
-      setIsClaimable(canClaim)
+      const claimable: Nft[] = []
```
Comment (ChefHutch) [inline src/views/Collectibles/components/GlobalCheckClaimStatus.tsx:31]: Modifications to `GlobalCheckClaimStatus` for the lottery NFTs

File: src/views/Collectibles/components/NftGiveawayModal.tsx [hunk 5]
```diff
@@ -32,25 +29,12 @@ const showConfetti = () => {
...
-  const identifier = Object.keys(teamNftMap).find(
-    (mapNftIdentifier) => teamNftMap[mapNftIdentifier] === profile.team.id,
-  )
-  return nftList.find((nft) => nft.identifier === identifier)
+interface NftGiveawayModalProps extends InjectedModalProps {
+  nfts: Nft[]
 }
 
-const NftGiveawayModal: React.FC<InjectedModalProps> = ({ onDismiss }) => {
+const NftGiveawayModal: React.FC<NftGiveawayModalProps> = ({ onDismiss, nfts }) => {
```
Comment (ChefHutch) [inline src/views/Collectibles/components/NftGiveawayModal.tsx:36]: Modifying the GiveAway modal to receive an array of nfts and render their images. Any `can user claim nft` logic is to be handled within the `GlobalCheckClaimStatus` hook

File: src/views/Profile/components/ClaimGiftModal.tsx [hunk 6]
```diff
@@ -22,8 +22,12 @@ export const useCanClaim = () => {
   useEffect(() => {
     const fetchClaimStatus = async () => {
       const claimRefundContract = getClaimRefundContract()
-      const walletCanClaim = await claimRefundContract.canClaim(account)
```
Comment (ChefHutch) [inline src/views/Profile/components/ClaimGiftModal.tsx:25]: This function throws an error on first load on **testnet** (the error is thrown on `develop` testnet too)- 

```
Unhandled Rejection (Error): call revert exception (method="canClaim(address)", errorArgs=null, errorName=null, errorSignature=null, reason=null, code=CALL_EXCEPTION, version=abi/5.4.0)
```

The error is handled safely by wrapping it in a try/catch.

File: src/config/constants/campaigns.ts [hunk 7]
```diff
@@ -180,6 +180,27 @@ const campaigns: Campaign[] = [
     description: 'Played a round of Prediction before round 12,120',
     badge: 'clairvoyant.svg',
   },
+  {
```
Comment (ChefHutch) [inline src/config/constants/campaigns.ts:183]: Note that the achievements are also being added in this PR

File: src/views/Collectibles/helpers.ts [hunk 8]
```diff
@@ -0,0 +1,137 @@
+import { getBunnySpecialLotteryContract } from 'utils/contractHelpers'
```
Comment (hachiojidev) [inline src/views/Collectibles/helpers.ts:1]: I split up the functions to get the 3 different claimable statuses here.

File: src/views/Collectibles/hooks/useBunnySpecialLottery.tsx [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+      const userParticipation = await getGraphLotteryUser(account, undefined, undefined, {
+        lottery_in: lotteryIdsArray,
+      })
+      setUserRounds(userParticipation.rounds)
+    }
+
+    if (account && currentLotteryIdNum && !userRounds) {
+      getUserData()
+    }
+  }, [currentLotteryIdNum, userRounds, account])
```
Comment (hachiojidev) [inline src/views/Collectibles/hooks/useBunnySpecialLottery.tsx:66]: Before this was two separate effects - however one was dependent on the other so I combined them for readability.

**Review by RabbitDoge — APPROVED** [review]

**Pull request merged**